### DISCUSSION TITLE
请问v1.x与v2.x兼容吗

### DISCUSSION THREAD

**Discussion by maoball** [1Panel-dev/1Panel#6199] [Q&A]
### 1Panel 版本

v1.10.13-lts

### 请描述您的需求或者改进建议

如题，可想到例如
1.v1.x能否直接升级到v2.x（是否需要卸载重装）
2.v1.x中安装的应用在v2.x中能否直接用
3.v1.x的备份v2.x能否恢复
谢谢开发者们😄

**Comment by wanghe-fit2cloud** [comment]
前两点我们会尽量做到兼容。
第三点，v1 版本的快照无法在 v2 版本中使用，但应用、网站、数据库等的备份我们会尽量实现兼容。

**Reply by maoball** [reply]
好的了解

**Comment by iygwei** [comment]
v2 在那有安装地址

**Reply by wan92hen** [reply]
还未发布

**Comment by NewTypo** [comment]
1panel 升级需要重新部署服务么，🥲小白不太懂，
v1中应用的备份，在v2中可以恢复么。感谢

**Reply by wanghe-fit2cloud** [reply]
1. V1 版本迁移至 V2 版本教程请参考：https://gitee.com/fit2cloud-feizhiyun/1panel-migrator
2. V1 和 V2 版本的备份、快照等不兼容

**Comment by masx200** [comment]
我发现如果把1panel的v1.30版本卸载后， 把1panel的v2直接安装到原来的1panel数据目录，1panel-core直接无法启动！

删除的时候就删掉了几个1panel的可执行文件，没有删掉数据目录

**Reply by wanghe-fit2cloud** [reply]
使用 `1pctl uninstall` 命令卸载 V1 版本时，数据目录不会被保留，因此理论上不会影响后续安装的 V2 版本。
为了更好地排查问题，麻烦提供一下详细日志信息，可以使用以下命令查看：
```bash
journalctl -u 1panel-core -exf
```

**Reply by masx200** [reply]
> 使用 `1pctl uninstall` 命令卸载 V1 版本时，数据目录不会被保留，因此理论上不会影响后续安装的 V2 版本。
> 为了更好地排查问题，麻烦提供一下详细日志信息，可以使用以下命令查看：
> ```bash
> journalctl -u 1panel-core -exf
> ```

我卸载的时候忘记删掉原来的数据目录了

**Reply by masx200** [reply]
> > 使用 `1pctl uninstall` 命令卸载 V1 版本时，数据目录不会被保留，因此理论上不会影响后续安装的 V2 版本。
> > 为了更好地排查问题，麻烦提供一下详细日志信息，可以使用以下命令查看：
> > ```bash
> > journalctl -u 1panel-core -exf
> > ```
> 
> 我卸载的时候忘记删掉原来的数据目录了

难道说必须使用1panel-migrator才能升级吗？

**Reply by wanghe-fit2cloud** [reply]
是的，需要迁移工具来实现。
